src/modules/transaction/usecase/service/transaction.service.ts

```diff
@@ -1,8 +1,45 @@
-import { Injectable } from '@nestjs/common';
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { TransactionRepository } from '../../adapter/repository/transaction.repository';
+import { Transaction } from '../../entity/transaction.model';
+import { TransactionOutput } from '../../adapter/dto/transaction.output';
+import { CategoryService } from 'src/modules/category/usecase/service/category.service';
 
 @Injectable()
 export class TransactionService {
-    getHello(): string {
-        return 'Hello World!';
+    constructor(private readonly transactionRepository: TransactionRepository, private readonly categoryService: CategoryService) {}
+
+    async createTransaction(authUserId: string, transaction: Transaction): Promise<void> {
+        await this.validateCategory(transaction);
```

Comment:
@PCalape Why do we need to validate the category here?

Reply:
Hello @arnellebalane, this is to ensure that we will not save a category id in the transactions table with a non existent category in the categories table and ensure proper link of the two tables via foreign key